### PR TITLE
Make queries traceable again

### PR DESCRIPTION
**Pull request by oli-obk**
This can't be tested without something along the lines of https://github.com/rust-lang/rust/pull/111924 unfortunately.

We could benchmark turning query tracing into an `info` level tracing statement, but let's get this fix landed first so we can actually debug properly again

### CODE REVIEW

File: compiler/rustc_query_impl/src/plumbing.rs [hunk 1]
```diff
@@ -265,14 +265,18 @@ macro_rules! hash_result {
...
     ([(separate_provide_extern) $($rest:tt)*][$tcx:expr, $name:ident, $key:expr]) => {{
-        if let Some(key) = $key.as_local_key() {
+        let ret = if let Some(key) = $key.as_local_key() {
             ($tcx.query_system.fns.local_providers.$name)($tcx, key)
         } else {
             ($tcx.query_system.fns.extern_providers.$name)($tcx, $key)
-        }
+        };
+        tracing::trace!(?ret);
+        ret
```
Comment (cjgillot) [inline compiler/rustc_query_impl/src/plumbing.rs:279]: Could you move this to the call site of `call_provider`?

**Review by cjgillot — COMMENTED** [review]
r=me with one nit.

TBH, I had trouble to use this debug tracing of queries: difficult to filter and drowning in some return values, HIR in particular. How do you use it?

**Comment by oli-obk** [comment]
> difficult to filter and drowning in some return values, HIR in particular. How do you use it?

Yea, return values can be annoying. I mostly log to a file and then just skip over the bad parts 🙃 


wrt filtering: I filter queries to specific arguments by filtering on the debug representation of the `key`: `RUSTC_LOG=[typeck{key=\.\*main\.\*}]`. Not sure if that is what you mean?

**Comment by oli-obk** [comment]
fwiw: I am looking into how to generally work with logs in a more reasonable scheme than "have a log file and grep through it". I want to start dumping json logs and then figure out some UI (possibly just a terminal UI for working with tracing logs specifically) that can actually fold nested calls instead of just indenting them

**Pull request merged**